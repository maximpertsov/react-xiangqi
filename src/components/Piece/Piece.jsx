import React from 'react';
import PropTypes from 'prop-types';
import redGeneralIcon from './50px-Xiangqi_black_side_General.svg.png';

const Piece = ({ name, icon }) => (
  <img className="Piece" alt={name} src={icon} />
);

Piece.propTypes = {
  icon: PropTypes.string,
  name: PropTypes.string,
};

Piece.defaultProps = {
  icon: redGeneralIcon,
  name: 'Red General',
};

export default Piece;
