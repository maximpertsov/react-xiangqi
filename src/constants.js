import { Color } from 'services/logic/constants';

export const AutoMove = Object.freeze({
  BLACK: Color.BLACK,
  BOTH: 'both',
  OFF: 'off',
  RED: Color.RED,
});

export default {};
