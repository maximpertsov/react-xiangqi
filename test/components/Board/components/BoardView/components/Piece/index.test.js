import React from 'react';
import { render } from 'enzyme';
import Piece from '../../../../../../../src/components/Board/components/BoardView/components/Piece';
import { ALL_PIECES } from '../../../../../../../src/services/logic/constants';

describe('Piece', () => {
  test.each(ALL_PIECES)('with code %s', code => {
    const wrapper = render(<Piece code={code} />);
    expect(wrapper).toMatchSnapshot();
  });

  test.each(ALL_PIECES)('moving with code %s', code => {
    const wrapper = render(<Piece code={code} moveX={1} moveY={2} />);
    expect(wrapper).toMatchSnapshot();
  });
});
