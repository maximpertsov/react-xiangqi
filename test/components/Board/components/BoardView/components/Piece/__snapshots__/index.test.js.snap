// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Piece moving with code A 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code B 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code C 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code K 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code N 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code P 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code R 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code a 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code b 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code c 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code k 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code n 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code p 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece moving with code r 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-rl2k4u",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code A 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code B 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code C 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code K 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code N 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code P 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code R 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code a 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code b 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code c 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code k 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code n 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code p 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Piece with code r 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "alt": "",
      "class": "Piece css-1aulbbe",
      "src": "[object Object]",
    },
    "children": Array [],
    "name": "img",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
    "x-attribsPrefix": Object {
      "alt": undefined,
      "class": undefined,
      "src": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
